src/tests/ftest/util/storage_utils.py

```diff
@@ -0,0 +1,555 @@
+            return self.address > other.address
+        if 'optane' in self.device.lower():
+            return True
+        return False
+
+    def __hash__(self):
+        """Get the hash value of this StorageDevice object.
+
+        Returns:
+            int: hash value of this StorageDevice object
+
+        """
+        return hash(str(self))
+
+    @property
+    def description(self):
+        """Get the StorageDevice description.
+
+        Returns:
+            str: the description of this StorageDevice
+
+        """
+        return ': '.join([self.storage_class, self.device])
+
+    @property
+    def is_backing(self):
+        """Is this device a backing device behind a VMD controller.
+
+        Returns:
+            bool: True if this device a backing device; False otherwise.
+
+        """
+        return self.is_disk and len(self.address.split(':')[0]) > 4
+
+    @property
+    def is_controller(self):
+        """Is this device a VMD controller.
+
+        Returns:
+            bool: True if this device an VMD controller; False otherwise.
+
+        """
+        return bool(self.managed_devices)
+
+    @property
+    def is_disk(self):
+        """Is this device a disk.
+
+        Returns:
+            bool: True if this device a disk; False otherwise.
+
+        """
+        return not self.is_controller
+
+
+def is_disk(device):
+    """Is this device a disk.
+
+    Args:
+        device (StorageDevice): the device to check
+
+    Returns:
+        bool: True if this device a disk; False otherwise.
+
+    """
+    return device.is_disk
+
+
+def is_controller(device):
+    """Is this device a VMD controller.
+
+    Args:
+        device (StorageDevice): the device to check
+
+    Returns:
+        bool: True if this device an VMD controller; False otherwise.
+
+    """
+    return device.is_controller
+
+
+class StorageInfo():
+    """Information about host storage."""
+
+    TIER_KEYWORDS = ['pmem']
+    TYPE_SEARCH = {
+        'NVMe': r"grep -E 'Class:\s+Non-Volatile memory controller' -B 1 -A 2",
+        'VMD': r"grep -E 'Device:\s+Volume Management Device NVMe RAID Controller' -B 2 -A 1",
+    }
+
+    def __init__(self, logger, hosts):
+        """Initialize a StorageInfo object.
+
+        Args:
+            logger (logger): logger for the messages produced by this class
+            hosts (NodeSet): set of hosts from which to obtain the storage device information
+        """
+        self._log = logger
+        self._hosts = hosts.copy()
+        self._devices = []
+
+    @property
+    def devices(self):
+        """Get the devices found on the hosts.
+
+        Returns:
+            list: a list of StorageDevice objects
+
+        """
+        return self._devices
+
+    @property
+    def disk_devices(self):
+        """Get the list of detected disk devices.
+
+        Returns:
+            list: a list of diskStorageDevice objects
+
+        """
+        return list(filter(is_disk, self.devices))
```

Comment:
To avoid wrapper functions such as `is_disk(device)` I'd recommend one of these approaches:
1. Using the getter for the peroperty
```suggestion
        return list(filter(StorageDevice.is_disk.fget, self.devices))
```
2. Using lambda magic
```suggestion
        return list(filter(lambda dev: dev.is_disk, self.devices))
```